velox/functions/prestosql/aggregates/MaxSizeForStatsAggregate.cpp

```diff
@@ -0,0 +1,269 @@
+      return;
+    }
+
+    auto numRows = arg->size();
+    std::vector<vector_size_t> elementSizes(numRows, 0);
+    std::vector<vector_size_t*> elementSizePtrs(numRows);
+    std::vector<IndexRange> elementIndices(numRows);
+
+    for (int i = 0; i < numRows; i++) {
+      elementIndices[i] = IndexRange{i, 1};
+      elementSizePtrs[i] = &elementSizes[i];
+    }
+
+    getVectorSerde()->estimateSerializedSize(
+        arg,
+        folly::Range(elementIndices.data(), numRows),
+        elementSizePtrs.data());
+
+    rows.applyToSelected([&](vector_size_t i) {
+      upsertOneAccumulator(groups[i], *baseVector, elementSizes, indices[i]);
+    });
+  }
+
+  void doUpdateSingleGroup(
+      char* group,
+      const SelectivityVector& rows,
+      const VectorPtr& arg) {
+    DecodedVector decoded(*arg, rows, true);
+    auto indices = decoded.indices();
+    auto baseVector = decoded.base();
+
+    auto numRows = baseVector->size();
+    std::vector<vector_size_t> elementSizes(numRows, 0);
+    std::vector<vector_size_t*> elementSizePtrs(numRows);
+    std::vector<IndexRange> elementIndices(numRows);
+
+    for (int i = 0; i < numRows; i++) {
+      elementIndices[i] = IndexRange{i, 1};
+      elementSizePtrs[i] = &elementSizes[i];
+    }
+
+    getVectorSerde()->estimateSerializedSize(
+        arg,
+        folly::Range(elementIndices.data(), numRows),
+        elementSizePtrs.data());
+
+    if (decoded.isConstantMapping()) {
+      if (decoded.isNullAt(0)) {
+        // nothing to do; all values are nulls
+        return;
+      }
+      upsertOneAccumulator(group, *baseVector, elementSizes, 0);
+      return;
+    }
+
+    rows.applyToSelected([&](vector_size_t i) {
+      upsertOneAccumulator(group, *baseVector, elementSizes, i);
+    });
+  }
+};
+
+bool registerMaxSizeForStatsAggregate(const std::string& name) {
+  // Types here are used by PlanBuilder to populate partial and final
+  // aggregation input and output types. E.g. intermediate results will be of
+  // type vector<intermediateType> and used to serve as the input vector type to
+  // final aggregation. argumentType is type of vector to AddRawInput
+  // intermediateType is the type of vector that extractAccumulator write to and
+  // that addIntermediateResults reads out of
+  // returnType is the type of vector that extractValues write to.
+  std::vector<std::shared_ptr<exec::AggregateFunctionSignature>> signatures;
+  for (const auto& inputType :
+       {"tinyint", "smallint", "integer", "bigint", "double", "real"}) {
+    // See implementation overview for the rationale of choosing these types.
+    signatures.push_back(
+        exec::AggregateFunctionSignatureBuilder()
+            .returnType(mapTypeKindToName(TypeKind::BIGINT))
+            .intermediateType(mapTypeKindToName(TypeKind::BIGINT))
+            .argumentType(inputType)
+            .build());
+  }
+
+  // For non-fixed width input types.
+  signatures.push_back(
```

Comment:
I think a single generic signature would be sufficient. By convention, use "T" for type variable.

```
exec::AggregateFunctionSignatureBuilder()
          .typeVariable("T")
          .returnType("bigint")
          .intermediateType("bigint")
          .argumentType("T")
          .build());
```